tests/test_pi_io.py

```diff
@@ -14,14 +14,30 @@ def test_turn_pin_on(self):
         pin = 1
         IO.turn_pin_on(pin)
         mock_GPIO.setup.assert_called_once_with(pin, mock_GPIO.OUT)
-        mock_GPIO.output.assert_called_once_with(pin, 1)
-        mock_GPIO.cleanup.assert_called_once_with(pin)
+        mock_GPIO.output.assert_called_once_with(pin, mock_GPIO.HIGH)
 
     def test_turn_pin_off(self):
         mock_GPIO = mock.Mock()
         IO = pi_io.IO(mock_GPIO)
         pin = 1
         IO.turn_pin_off(pin)
         mock_GPIO.setup.assert_called_once_with(pin, mock_GPIO.OUT)
-        mock_GPIO.output.assert_called_once_with(pin, 0)
-        mock_GPIO.cleanup.assert_called_once_with(pin)
\ No newline at end of file
+        mock_GPIO.output.assert_called_once_with(pin, mock_GPIO.LOW)
+
+    def test_setup_only_called_once_per_pin(self):
+        mock_GPIO = mock.Mock()
+        IO = pi_io.IO(mock_GPIO)
+        pin_one = 1
+        pin_two = 2
+        IO.turn_pin_on(pin_one)
+        IO.turn_pin_on(pin_two)
+        IO.turn_pin_off(pin_one)
+        IO.turn_pin_on(pin_one)
+        IO.turn_pin_off(pin_two)
+        IO.turn_pin_on(pin_two)
+
+        self.assertEqual(mock_GPIO.setup.call_args_list,
+                         [((pin_one, mock_GPIO.OUT),),
+                          ((pin_two, mock_GPIO.OUT),)])
+        pin_list = [pin_one, pin_two]
+        self.assertEqual(IO._output_pins, set(pin_list))
```

Comment:
Suggest cutting 42-43.

39-41 gets us sufficient confidence that the class did the right thing. We're interested in whether `IO` correctly calls the `GPIO` library. The `_output_pins` is an implementation detail that the test should ideally not peek into.